src/Canvas.cc

```diff
@@ -57,17 +57,21 @@ Canvas::Initialize(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE target) {
   ctor->SetClassName(Nan::New("Canvas").ToLocalChecked());
 
   // Prototype
-  Local<ObjectTemplate> proto = ctor->PrototypeTemplate();
+  Local<ObjectTemplate> proto = ctor->InstanceTemplate();
   Nan::SetPrototypeMethod(ctor, "toBuffer", ToBuffer);
   Nan::SetPrototypeMethod(ctor, "streamPNGSync", StreamPNGSync);
   Nan::SetPrototypeMethod(ctor, "streamPDFSync", StreamPDFSync);
 #ifdef HAVE_JPEG
   Nan::SetPrototypeMethod(ctor, "streamJPEGSync", StreamJPEGSync);
 #endif
-  SetProtoAccessor(proto, Nan::New("type").ToLocalChecked(), GetType, NULL, ctor);
-  SetProtoAccessor(proto, Nan::New("stride").ToLocalChecked(), GetStride, NULL, ctor);
-  SetProtoAccessor(proto, Nan::New("width").ToLocalChecked(), GetWidth, SetWidth, ctor);
-  SetProtoAccessor(proto, Nan::New("height").ToLocalChecked(), GetHeight, SetHeight, ctor);
+
+  // link getters and setter to the object property
+  Local<ObjectTemplate> inst = ctor->InstanceTemplate();
+
+  Nan::SetAccessor(inst, Nan::New("type").ToLocalChecked(), GetType, NULL);
+  Nan::SetAccessor(inst, Nan::New("stride").ToLocalChecked(), GetStride, NULL);
+  Nan::SetAccessor(inst, Nan::New("width").ToLocalChecked(), GetWidth, SetWidth);
+  Nan::SetAccessor(inst, Nan::New("height").ToLocalChecked(), GetHeight, SetHeight);
```

Comment:
Thanks for the PR! I've only skimmed this, but I think these properties are supposed to be on the prototype, not the instance. That's at least how Firefox and Chromium implement it.

Reply:
@zbjornson I was not aware of that. After some investigation the problem seems to be this test case `
   assert.throws(function () { Canvas.prototype.width }, /incompatible receiver/)` since it causes the getter to be invoked with an invalid `this` context. When adding the same if check you applied here https://github.com/Automattic/node-canvas/pull/1550 for height and width all test cases are passing again.